source/agora/test/ValidatorRecurringEnrollment.d

```diff
@@ -234,44 +234,47 @@ unittest
     auto blocks = node_1.getBlocksFrom(0, 2);
     assert(blocks.length == 1);
 
-    auto sleep_node_1 = nodes[$ - 1];
-    auto sleep_node_2 = nodes[$ - 2];
+    auto sleep_node_1 = nodes[$ - 1]; // node #5
+    auto sleep_node_2 = nodes[$ - 2]; // node #4
 
     // Approach end of the cycle
-    network.generateBlocks(Height(GenesisValidatorCycle - 2));
+    network.generateBlocks(Height(GenesisValidatorCycle - 3));
 
     // Make 2 nodes sleep
     sleep_node_1.ctrl.sleep(60.seconds, true);
     sleep_node_2.ctrl.sleep(60.seconds, true);
-
     network.generateBlocks(iota(GenesisValidators - 2),
         Height(GenesisValidatorCycle - 1));
 
-    // Wake one up right before cycle ends
+    // Wake one up right before cycle ends and set recurringEnrollment
     sleep_node_2.ctrl.sleep(0.seconds);
+
     // Let it catch up
-    network.expectHeightAndPreImg(iota(0, GenesisValidators - 1),
+    network.expectHeightAndPreImg(only(GenesisValidators - 2), // node #4
         Height(GenesisValidatorCycle - 1), network.blocks[0].header);
 
-    network.generateBlocks(iota(GenesisValidators - 1),
+    network.generateBlocks(iota(GenesisValidators - 1), // nodes #0 -> #4
         Height(GenesisValidatorCycle));
 
-    blocks = node_1.getBlocksFrom(10, GenesisValidatorCycle + 3);
+    blocks = node_1.getBlocksFrom(GenesisValidatorCycle, 1);
     auto enrolls1 = blocks[$ - 1].header.enrollments.length;
 
-    // This nodes will wake up to an expired cycle, it should immediately enroll
+    assert(enrolls1 == GenesisValidators - 1);
```

Comment:
This is not as expected. All validators are enrolled!

Reply:
Thanks @omerfirmak your latest PR fixed this.

Reply:
I spoke too soon. It can still fail on this check. So it seems that the validator can be re-enroll even when sleeping.

Reply:
Yeah, as discussed yesterday, `sleep` only makes the server-side sleep... It does nothing to prevent the timers from firing.

Reply:
Updated to skip `checkAndEnroll` in this test during the time the nodes are put to sleep.

Reply:
That's solved AFAIK.